gstvstaudioprocessor.cpp

```diff
@@ -1225,18 +1225,91 @@ list_paths_with_vst3_extension (VST3::Hosting::Module::PathList &paths,
   g_dir_close (dir);
 }
 
+static gboolean
+compare_versions (const gchar *version_str, const gchar *version_req_str)
+{
+  gchar op[3];
+  gint major = 0;
+  gint minor = 0;
+  gint subminor = 0;
+  gint subsubminor = 0;
+  gint major_req = 0;
+  gint minor_req = 0;
+  gint subminor_req = 0;
+  gint subsubminor_req = 0;
+  gint version = 0;
+  gint version_req = 0;
+  gint n_parts = sscanf(version_str, "%d.%d.%d.%d", &major, &minor, &subminor, &subsubminor);
+  gint n_parts_req = sscanf(version_req_str, "%2[<>=]%d.%d.%d.%d", op, &major_req, &minor_req, &subminor_req, &subsubminor_req);
+
+  if (n_parts > 0)
+    version = (major & 0xff) << 24;
+  if (n_parts > 1)
+    version += (minor & 0xff) << 16;
+  if (n_parts > 2)
+    version += (subminor & 0xff) << 8;
+  if (n_parts > 3)
+    version += (subsubminor & 0xff);
+
+  if (n_parts_req > 1)
+    version_req = (major_req & 0xff) << 24;
```

Comment:
+= for consistency I guess? Is each component guaranteed to be 0<=X<=255?

Reply:
I could not find documentation about plugins' version numbers, but this is based on some code in the VST3_SDK (`source/vst/vst2wrapper/vst2wrapper.cpp:2087:void Vst2Wrapper::setEffectVersion (char* version)`)